test/org/sagebionetworks/bridge/dynamodb/DynamoNotificationTopicDaoTest.java

```diff
@@ -387,44 +264,102 @@ public void deleteTopicPermanentlyWithCriteria() {
+
+        // Execute and verify.
+        dao.listTopics(TEST_STUDY, false);
+
+        // Verify query.
+        verify(mockMapper).queryPage(eq(DynamoNotificationTopic.class), queryExpressionCaptor.capture());
         
-        request = deleteTopicRequestCaptor.getAllValues().get(1);
-        assertEquals(topic2.getTopicARN(), request.getTopicArn());
+        // Query expression does include filter of deleted items.
+        DynamoDBQueryExpression<DynamoNotificationTopic> capturedQuery = queryExpressionCaptor.getValue();
+        assertEquals("{AttributeValueList: [{N: 1,}],ComparisonOperator: NE}",
+                capturedQuery.getQueryFilter().get("deleted").toString());
     }
     
+    @Test
+    public void listTopicsIncludeDeleted() {
+        // Mock DDB to return topics.
+        DynamoNotificationTopic topicWithoutCriteria = (DynamoNotificationTopic) getNotificationTopic();
+        topicWithoutCriteria.setGuid(GUID_WITHOUT_CRITERIA);
+        topicWithoutCriteria.setCriteria(null);
+
+        DynamoNotificationTopic topicWithCriteria = (DynamoNotificationTopic) getNotificationTopic();
+        topicWithCriteria.setGuid(GUID_WITH_CRITERIA);
+        topicWithCriteria.setCriteria(null);
+
+        mockListCall(topicWithoutCriteria, topicWithCriteria);
+
+        // Execute and verify.
+        List<NotificationTopic> topics = dao.listTopics(TEST_STUDY, true);
+        assertEquals(2, topics.size());
+
+        assertEquals(GUID_WITHOUT_CRITERIA, topics.get(0).getGuid());
+        assertNull(topics.get(0).getCriteria());
+
+        assertEquals(GUID_WITH_CRITERIA, topics.get(1).getGuid());
+        assertCriteria(GUID_WITH_CRITERIA, topics.get(1).getCriteria());
+
+        // Verify query.
+        verify(mockMapper).queryPage(eq(DynamoNotificationTopic.class), queryExpressionCaptor.capture());
+        
+        DynamoDBQueryExpression<DynamoNotificationTopic> capturedQuery = queryExpressionCaptor.getValue();
+        DynamoNotificationTopic capturedTopic = capturedQuery.getHashKeyValues();
+        assertEquals(TEST_STUDY_IDENTIFIER, capturedTopic.getStudyId());
+        assertNull(capturedTopic.getGuid());
+        assertNull(capturedQuery.getQueryFilter()); // deleted are not being filtered out
+    }
+
+    // Helper method to mock the list API.
+    private void mockListCall(DynamoNotificationTopic... topics) {
+        List<DynamoNotificationTopic> topicList = ImmutableList.copyOf(topics);
+        when(mockQueryResultPage.getResults()).thenReturn(topicList);
+        when(mockMapper.queryPage(eq(DynamoNotificationTopic.class), any())).thenReturn(mockQueryResultPage);
+    }
     
     // If the SNS topic creation fails, no DDB record will exist.
     // (okay if SNS topic exists, but DDB record doesn't, so don't need to test that path)
     @Test
     public void noOrphanOnPartialCreate() {
-        doThrow(new RuntimeException()).when(mockSnsClient).createTopic(any(CreateTopicRequest.class));
-        
```

Comment:
This wasn't called and then a NPE was making the test succeed (it wasn't called because the method is overloaded and it's looking for the wrong method overload). Restored, fixed overload so it is called, test passes.